doc/api/esm.md

```diff
@@ -1188,6 +1188,39 @@ export async function transformSource(source,
 }
 ```
 
+#### <code>getGlobalPreloadCode</code> hook
+
+> Note: The loaders API is being redesigned. This hook may disappear or its
+> signature may change. Do not rely on the API described below.
+
+Sometimes it can be necessary to run some code inside of the same global scope
+that the application will run in. This hook allows to return a string that will
+be ran as sloppy-mode script on startup.
+
+Similar to how CommonJS wrappers work, the code runs in an implicit function
+scope. The only argument is a `require`-like function that can be used to load
+builtins: `getBuiltin(request: string)`.
```

Comment:
By `argument` you mean, function-scope variable made available within this implicit wrapper? So the CommonJS magic not-really-globals are _not_ available?

```suggestion
Similar to how CommonJS wrappers work, the code runs in an implicit function
scope; but the CommonJS wrapper variables such as `require` are not available.
The only implicit scoped variable available is a new one for loading Node.js
core modules: `getBuiltin(request: string)`. It functions similarly to
`require`, but can only be used for Node.js core modules such as `'fs'`.
```

Reply:
I think using a different term when explaining `getBuiltin` ("for loading core modules") is a bit confusing. But +1 to adding an example module.

Reply:
Feel free to revise. My concern was that I'm not sure average users understand what a builtin is. I looked for a link in the docs for Node core modules but couldn't find that either; it's the list of sections on the left, clearly, but I feel like there must be a place that lists all the builtins and how they differ from NPM packages.

Anyway I'm okay with `getBuiltin`; I was trying to think of something better (`requireNodeCoreModule`?) but I'm not sure I have a better suggestion.

Reply:
I added an example ("fs") but I think the more important piece when it comes to `require` is what's in the next paragraph: How to get it when the code needs it. I'd rather have people quickly discover that than reiterate that they don't have it by default. :)

Reply:
I think the kinds of people who are likely to use this feature would have an idea what `getBuiltin` might do. We do use the name in our API, e.g. `require('module').builtinModules` is where we list the names of the builtin module names.